src/cdk/v2/destinations/the_trade_desk/transformConversion.js

```diff
@@ -0,0 +1,88 @@
+const { InstrumentationError } = require('@rudderstack/integrations-lib');
+const {
+  defaultRequestConfig,
+  simpleProcessRouterDest,
+  defaultPostRequestConfig,
+  removeUndefinedAndNullValues,
+} = require('../../../../v0/util');
+const { EventType } = require('../../../../constants');
+const { REAL_TIME_CONVERSION_ENDPOINT } = require('./config');
+const {
+  prepareFromConfig,
+  prepareCommonPayload,
+  getRevenue,
+  prepareItemsPayload,
+  getAdvertisingId,
+  prepareCustomProperties,
+  populateEventName,
+  getDataProcessingOptions,
+  getPrivacySetting,
+  enrichTrackPayload,
+} = require('./utils');
+
+const responseBuilder = (payload) => {
+  const response = defaultRequestConfig();
+  response.endpoint = REAL_TIME_CONVERSION_ENDPOINT;
+  response.method = defaultPostRequestConfig.requestMethod;
+  response.body.JSON = payload;
+  return response;
+};
+
+const validateInput = (message) => {
+  if (!message.type) {
+    throw new InstrumentationError('Event type is required');
+  }
+
+  const messageType = message.type.toLowerCase();
+  if (messageType !== EventType.TRACK) {
+    throw new InstrumentationError(`Event type "${messageType}" is not supported`);
+  }
+
+  if (!message.event) {
+    throw new InstrumentationError('Event is not present. Aborting.');
+  }
+};
+
+const prepareTrackPayload = (message, destination) => {
+  const configPayload = prepareFromConfig(destination);
+  const commonPayload = prepareCommonPayload(message);
+  const items = prepareItemsPayload(message);
+  const { id, type } = getAdvertisingId(message);
+  const customProperties = prepareCustomProperties(message, destination);
+  const eventName = populateEventName(message, destination);
+  const value = getRevenue(message);
+  let payload = {
+    ...configPayload,
+    ...commonPayload,
+    event_name: eventName,
+    value,
+    items,
+    adid: id,
+    adid_type: type,
+    ...customProperties,
+    data_processing_option: getDataProcessingOptions(message),
+    privacy_settings: getPrivacySetting(message),
+  };
+
+  payload = enrichTrackPayload(message, payload);
```

Comment:
what are the customFields extraction happening again in enrichTrackPayload ?

Reply:
Added the function description. 
`enrichTrackPayload` enriches the track payload with extra properties present in 'properties' other than the ones defined in TTDCommonConfig.json and TTDItemConfig.json